src/main/java/cn/RecipeAPI/Models/Recipe.java

```diff
@@ -71,12 +82,12 @@ public void validate() throws IllegalStateException {
         }
     }
 
-    @PostLoad
+    //    @PostPersist  // Cannot be used during testing.
```

Comment:
Based on the tests, and structure they are integration tests of the controller's layer, where you mock the service layer. 
The `PostPersist` hook is triggered after saving DB. Gut since the service late is mocked it makes it to DB. 

If the logic happens beyond the test boundary it should be part of the mocking. In a real application, the service would return an entity with the location populated, so the mock should do the same.